common/djangoapps/student/views/management.py

```diff
@@ -172,6 +176,23 @@ def index(request, extra_context=None, user=AnonymousUser()):
     return render_to_response('index.html', context)
 
 
+def show_auto_generated_username(username):
+    """
+    Check if the auto-generated username is valid based on the specified pattern.
+
+    Parameters:
+    - username (str): The username to be checked.
+
+    Returns:
+    - bool: True if the username is valid and the auto-generated username check is enabled, False otherwise.
+    """
+    if not is_auto_generated_username_enabled():
+        return False
+
+    pattern = r'^[A-Z]{1,2}_\d{4}_[A-Z0-9]+$'
```

Comment:
Have you tested this regex with an auto-generated username?

Reply:
yes.